components/concept/sync/src/main/java/mozilla/components/concept/sync/Sync.kt

```diff
@@ -42,6 +42,24 @@ data class SyncAuthInfo(
     val tokenServerUrl: String
 )
 
+/**
+ * An extension of [SyncableStore] that can be locked/unlocked using an encryption key.
+ */
+interface LockableStore : SyncableStore {
+    /**
+     * Unlocks underlying storage using [encryptionKey]. Storage must be [ensureLocked] after use.
```

Comment:
nit: If we always should call `ensureLocked()` after use then could we design this API to always do that. e.g. something like:

```Kotlin
store.unlocked(encryptionKey) {
  // here is is unlocked and after this block it gets locked automatically
}
```

which internally does something like
```
try {
  store.ensureUnlocked(encryptionKey)
  block(store)
} finally {
  store.ensureLocked()
}
```

Just to prevent misuse? But I do not know enough about how this is used to know whether that makes sense.

Reply:
That is much better. Fixed.